docs/5.x/terraform.md

```diff
@@ -0,0 +1,534 @@
+    - github: Use github as an identity provider.
+* `second_factor` - Whether to enable second factor authentication when using local identity provider.
+    - off: Second factor authentication is disabled on the cluster.
+    - otp: Use TOTP based second factor authentication.
+    - u2f: Use U2F based second factor authentication.
+* `connector_name` - (Optional) The name of the OIDC or SAML connector to use for providing identity. If left blank, the first provider will be used.
+* `u2f_appid` - (Optional) The application ID of the cluster. See [the teleport documents](https://gravitational.com/teleport/docs/admin-guide/#fido-u2f) for more information.
+* `u2f_facets` - (Optional) A list of facets for U2F authentication. See [the teleport documents](https://gravitational.com/teleport/docs/admin-guide/#fido-u2f) for more information.
+
+## gravity_github
+Configures the cluster to allow authentication using github as an identity provider.
+
+### Example Usage
+```
+resource "gravity_github" "test" {
+  name          = "github"
+  display       = "Github"
+  client_id     = "<client-id>"
+  client_secret = "<client-secret>"
+  redirect_url  = "https://<cluster-url>/portalapi/v1/github/callback"
+
+  teams_to_logins {
+    organization = "example"
+    team         = "admins"
+    logins       = ["@teleadmin"]
+  }
+}
+```
+
+### Argument Reference
+The following arguments are supported:
+
+* `name` - The name of the connector. This name must be unique.
+* `client_id` - Github OAuth app client ID to use.
+* `client_secret` - Github OAuth app client secret.
+* `redirect_url` - URL that the cluster can be reached at for OAuth callback.
+* `display` - Human readable display name that will be presented to users on the web interface when logging in.
+* `teams_to_logins` - One or more maps of organization/team/logins to allow on the cluster.
+    - organization - The github organization a user belongs to.
+    - team - The team within the organization that the user belongs to.
+    - logins - A list of allowed logins for this organization/team on the cluster.
+
+## gravity_log_forwarder
+Configure log forwarding to an external syslog server.
+
+### Example Usage
+```
+resource "gravity_log_forwarder" "test" {
+  name     = "logzer"
+  address  = "192.168.1.1:514"
+  protocol = "udp"
+}
+```
+
+### Argument Reference
+The following arguments are supported:
+
+* `name` - A name to use for the forwarder.
+* `address` - The IP address or hostname and port to send logs to in the format `<host>:<port>`.
+* `protocol` - Which transpor protocol to use for log forwarding.
+    - tcp - Use TCP transport.
+    - udp - Use UDP transport.
+
+## gravity_tlskeypair
+Apply a TLS Certificate and Key to the cluster to be used for the Web UI and API of the cluster.
+
+### Example Usage
+```
+resource "gravity_tlskeypair" "test" {
+  cert = <<EOF
+-----BEGIN CERTIFICATE-----
+MIIGGDCCBQCgAwIBAgISA/wgCy4eFgnVgvgm4TGN4CCwMA0GCSqGSIb3DQEBCwUA
```

Comment:
this looks like a valid certificate - is it being used anywhere?

Reply:
No, it's just a cert off a test cluster to show what it looks like, but thinking about it, I'll cut it down just so it isn't as visually overwhelming.